discord_connector.py

```diff
@@ -413,6 +466,70 @@ def run_in_loop(self, coroutine, action_result, message=""):
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        if not self.is_poll_now():
+            self.debug_print("Maximum number of containers have been reached")
+            return action_result.set_status(phantom.APP_ERROR, "Maximum number of containers have been reached")
+
+        container_count = param["container_count"]
+
+        self._status = self.load_state()
+        last_poll = self._state.get("last poll", None)
+        if last_poll is not None:
+            _, last_poll = self.parse_date(last_poll)
+        self._state["last poll"] = str(datetime.now().strftime("%Y-%m-%d %H:%M:%S"))
+        self.save_state(self._state)
+
+        status, channels = self.run_in_loop(self._guild.fetch_channels(), action_result,
+                                            message="Cannot fetch channel from Discord.")
+        if not status:
+            self.debug_print("action result: Cannot Poll messages, unable to fetch channels ")
+            return action_result.set_status(phantom.APP_ERROR, "action result: Cannot Poll messages")
+
+        for channel in channels:
+            if isinstance(channel, discord.TextChannel):
+                self.save_progress("checking channel: {} | {}".format(channel.name, channel.id))
+
+                status, messages = self.fetch_message_history(channel.id, action_result, last_poll, None, None, container_count)
+                if not status:
+                    self.debug_print("action result: Cannot Poll messages from {} {}".format(channel.name, channel.id))
+                    return action_result.set_status(phantom.APP_ERROR, "action result: Cannot Poll messages")
+
+                self.save_progress("saving the containers")
+                for message in messages:
+                    contains_embeds_or_attachments = True if (message.attachments or message.embeds) else False
+
+                    container = {
+                        "name": f"message {message.id} on channel {channel.name}",
+                        "description": "generated by on poll action",
+                        "sensitivity": "White" if not contains_embeds_or_attachments else "Green"
+                    }
+
+                    ret_val, ret_message, container_id = self.save_container(container)
+                    self.save_progress(
+                        "return value: {} | message: {} | container_id: {}".format(ret_val, ret_message, container_id))
+                    if phantom.is_fail(ret_val):
+                        return action_result.set_status(
+                            phantom.APP_ERROR,
+                            "Unable to create container: {}".format(message)
+                        )
+
+                    # this data allows us to proceed with further actions yet does not give any insight, should we consider
+                    # extending it?
+                    artifact = Artifact(
```

Comment:
I think that artifact should contain message content and attachments

Reply:
artifact now contains also message content